File: LeetCode/[M] - 901. Online Stock Span/solution.py
class StockSpanner:
    def __init__(self):
        self.stack = [(-1, 0)]

    def next(self, price: int) -> int:
        days = 1
        while self.stack and price >= self.stack[-1][0]:
            days += self.stack[-1][1]
            self.stack.pop()
        self.stack.append((price, days))
        return days


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: LeetCode/[M] - 901. Online Stock Span/solution.go
package main

type StockSpanner struct {
	stack [][]int
}

func Constructor() StockSpanner {
	return StockSpanner{stack: make([][]int, 0)}
}

func (ss *StockSpanner) Next(price int) int {
	days := 1
	for len(ss.stack) != 0 && price >= ss.stack[len(ss.stack)-1][0] {
		days += ss.stack[len(ss.stack)-1][1]
		ss.stack = ss.stack[:len(ss.stack)-1]
	}
	ss.stack = append(ss.stack, []int{price, days})
	return days
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Next(price);
 */
